server/topic.go

```diff
@@ -1568,10 +1570,9 @@ func (t *Topic) thisUserSub(sess *Session, pkt *ClientComMessage, asUid types.Ui
 					"off+dis", nilPresParams, "", false)
 			}
 		}
-
-		// Apply changes.
-		t.perUser[asUid] = userData
 	}
+	// Apply changes.
+	t.perUser[asUid] = userData
```

Comment:
Aren't we supposed to avoid keeping perUserData for channel subscriptions in memory?
IIRC, the initial idea was to keep it only for owners and banned users b/c of memory issues.

Reply:
Yes, that was the original plan. But then I implemented the clients it did not look optimal anymore: it generated extra DB calls for trivial things like marking messages as read/received. I refactored it to keep the `perUserData` for the duration of the session. It looks like a reasonable compromise.

Reply:
Okay, makes sense. So the only difference is the topic's `perUser` get loaded lazily.

Reply:
> So the only difference is the topic's perUser get loaded lazily.

Lazy load and cleared when all user's sessions disconnect.